homeassistant/components/matrix/__init__.py

```diff
@@ -67,6 +72,7 @@
 SERVICE_SCHEMA_SEND_MESSAGE = vol.Schema(
     {
         vol.Required(ATTR_MESSAGE): cv.string,
+        vol.Optional(ATTR_DATA): dict,
```

Comment:
We no longer allow changes to the YAML configuration for integrations that communicate with devices or services.

Instead, this integration needs to be converted to a configuration flow first.

For more information see: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started with that: <https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

Thanks 👍

Reply:
I will try to do so ASAP.

 Just to be sure, the yaml configuration modification will be kept since it's concern the notification sending process and not the configuration of the service (ie connection to the server), right?

Reply:
I have tried to add the config_flow for matrix. I have two problems : 
- the configuration is too complex to be defined without yaml support, the only parts that can be configured in config_flow will be the connection parameters
- if I'm not wrong the use of config_flow imply the conversion to async for the setup. This is quite a big change

Since the current PR modify the yaml but only in the notification part and not the server configuration, could it be accepted as it is ? I will still work on the config_flow integration in a new PR.

Reply:
We cannot merge this PR until there is a config flow and this functionality is implemented via an options flow. We don't accept any changes to YAML config anymore.

Reply:
This is not a change to YAML config. In fact, it has literally nothing to do with config.

Reply:
Oh. Good point. I had only looked at the comments, not the code. 

This change is fine.